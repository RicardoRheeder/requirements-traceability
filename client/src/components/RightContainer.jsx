import React from 'react'
import NotificationCard from './NotificationCard'
import { useSelector } from 'react-redux'

export default function RightContainer() {
  const notifications = useSelector((state) => state.user.notifications)

  const renderRecentNotifications = (notifications) => {
    console.log(notifications)
    if (notifications != null && Array.isArray(notifications)) {
      return notifications.map((notification, i) => {
        return <NotificationCard notification={notification} key={i} />
      })
    }
    return (
      <div className="left-container-placeholder">
        <h2>No existing notifications.</h2>
      </div>
    )
  }

  return (
    <div className="right-container">
      <h1>Notifications: </h1>
      {renderRecentNotifications(notifications)}
    </div>
  )
}
