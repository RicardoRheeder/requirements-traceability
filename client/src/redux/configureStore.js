import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

// restricting extension in production using process.env.NODE_ENV which is set 'production'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction';

import store from './stores';

const enhancer = (process.env.NODE_ENV == 'development')
               ? composeWithDevTools(applyMiddleware(thunk))
               : applyMiddleware(thunk);

export default () => createStore(store, enhancer);
