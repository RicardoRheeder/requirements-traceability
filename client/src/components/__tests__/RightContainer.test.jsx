import React from "react"
import Enzyme, { shallow, mount} from "enzyme"
import RightContainer from "../RightContainer"
import Adapter from "enzyme-adapter-react-16"
import { Provider } from "react-redux";
import configureStore from "redux-mock-store"


const mockStore = configureStore([])
Enzyme.configure({adapter: new Adapter() });

describe("RightContainer", () => {
    let store;
    beforeEach(() => {
        store = mockStore({
            document: {
                documents: [{title: 'testdoc', versions: [], tree: null}],
                versions: []
            },
            common: {
                selectedDocumentPanelObject: {title: 'testdoc'}
            },
            user: {
                notifications: []
            }
            
        });
    });
    test("renders", () => {
        const wrapper = shallow(<Provider store = {store}><RightContainer/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
    test("renders with children", () => {
        const wrapper = mount(<Provider store = {store}><RightContainer/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});