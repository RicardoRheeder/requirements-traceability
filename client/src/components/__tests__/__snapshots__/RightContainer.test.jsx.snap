// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightContainer renders 1`] = `ShallowWrapper {}`;

exports[`RightContainer renders with children 1`] = `ReactWrapper {}`;
